client/components/select-vertical/index.tsx

```diff
@@ -1,41 +1,75 @@
 import { useTranslate } from 'i18n-calypso';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useDebounce } from 'use-debounce';
 import FormLabel from 'calypso/components/forms/form-label';
-import { useSiteVerticalsQuery } from 'calypso/data/site-verticals';
+import { useSiteVerticalQuery, useSiteVerticalsQuery } from 'calypso/data/site-verticals';
 import SuggestionSearch from './suggestion-search';
 import type { Vertical } from './types';
 import type { SiteVerticalsResponse } from 'calypso/data/site-verticals';
 
 interface Props {
+	defaultVertical?: string;
 	isSkipSynonyms?: boolean;
 	onSelect?: ( vertical: Vertical ) => void;
 }
 
-const SelectVertical: React.FC< Props > = ( { isSkipSynonyms, onSelect } ) => {
+const SelectVertical: React.FC< Props > = ( { defaultVertical, isSkipSynonyms, onSelect } ) => {
 	const translate = useTranslate();
-	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ searchTerm, setSearchTerm ] = useState< string >( '' );
+	const [ isLoading, setIsLoading ] = useState< boolean >( false );
+	const [ isDisableInput, setIsDisableInput ] = useState< boolean >( false );
 	const [ debouncedSearchTerm ] = useDebounce( searchTerm, 300 );
+	const isQueryDefaultValue = typeof defaultVertical === 'string' && defaultVertical !== '';
+	const isQuerySuggestions = debouncedSearchTerm !== '';
+
+	const { data: defaultValue } = useSiteVerticalQuery(
+		{ id: defaultVertical },
+		{ enabled: isQueryDefaultValue }
+	);
+
 	const { data: suggestions } = useSiteVerticalsQuery(
 		{ term: debouncedSearchTerm, skip_synonyms: isSkipSynonyms },
-		{ enabled: '' !== debouncedSearchTerm }
+		{ enabled: isQuerySuggestions }
 	);
 
-	const mapSiteVerticalsResponseToVertical = ( verticals: SiteVerticalsResponse[] ): Vertical[] =>
-		verticals.map( ( vertical: SiteVerticalsResponse ) => ( {
-			value: vertical.id,
-			label: vertical.title,
-			category: String( translate( 'Suggestions' ) ),
-		} ) );
+	const mapOneSiteVerticalsResponseToVertical = ( vertical: SiteVerticalsResponse ): Vertical => ( {
```

Comment:
I think these mapping functions should be moved to the `select` property passed to useQuery. e.g. see [use-billing-dashboard-query.ts#L79](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/partner-portal/licenses/hooks/use-billing-dashboard-query.ts#L79)

Reply:
@roo2 that was the original approach, but @fushar brought up a good point [here](https://github.com/Automattic/wp-calypso/pull/62365#discussion_r842301998) in that it would be better to separate the shape of the Vertical UI component from the data layer. What do you think?

Reply:
I think we could use `select` from queryOption. But I'm okay with both.

Reply:
ahh ok, hmm that does make sense. It just seems more clunky to have it done in several steps. But I think that's a good point and happy with either way as well